cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -181,9 +185,22 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			},
 		}
 
-		if vmSpec.Zone != "" {
-			zones := []string{vmSpec.Zone}
-			virtualMachine.Zones = &zones
+		// Set availability set if no failure domains are available
+		if len(s.Scope.FailureDomains()) == 0 {
```

Comment:
this could be 

```
if vmSpec.Zone != "" {
 // set zone
} else {
 // set availability set
}
```

Since `len(s.Scope.FailureDomains()) != 0` <==> `vmSpec.Zone != ""`